packages/manager/apps/web-hosting/src/pages/websites/Cells.component.tsx

```diff
@@ -0,0 +1,263 @@
+    webSiteItem?.currentState.hosting.serviceName || '',
+    true,
+  );
+
+  const {
+    data,
+    isLoading,
+    isError,
+    refetch,
+  } = useWebHostingAttachedDomaindigStatus(
+    webSiteItem?.currentState.hosting.serviceName,
+    webSiteItem?.currentState.fqdn,
+  );
+
+  if (isError) {
+    return (
+      <OdsButton
+        label=""
+        icon={ODS_ICON_NAME.refresh}
+        variant={ODS_BUTTON_VARIANT.ghost}
+        onClick={() => refetch()}
+      />
+    );
+  }
+  return ['A', 'AAAA'].map((type) => {
+    let status = DnsStatus.NOT_CONFIGURED;
+    let recommendedIps: string[] = [];
+    let records: any[] = [];
+
+    if (!isLoading && data) {
+      records = Object.values(data.records).filter((r) => r.type === type);
+      recommendedIps =
+        type === 'A'
+          ? data.recommendedIps?.recommendedIpV4 || []
+          : data.recommendedIps?.recommendedIpV6 || [];
+
+      if (records.length > 0) {
+        const hasValidRecord = records.some((record) => {
+          if (record.dnsConfigured && record?.isOvhIp) {
+            const recordIp = Object.keys(data.records).find(
+              (ip) =>
+                data.records[ip].type === type && data.records[ip] === record,
+            );
+            return recordIp && recommendedIps.includes(recordIp);
+          }
+          return false;
+        });
+
+        if (hasValidRecord) {
+          status = DnsStatus.CONFIGURED;
+        } else {
+          const hasExternalRecord = records.some(
+            (record) => record?.isOvhIp === false,
+          );
+          status = hasExternalRecord
+            ? DnsStatus.EXTERNAL
+            : DnsStatus.NOT_CONFIGURED;
+        }
+      }
+    }
+
+    return (
+      <BadgeStatus
+        key={type}
+        itemStatus={status}
+        label={type}
+        isLoading={isLoading}
+        href={hostingUrl}
+      />
+    );
+  });
+};
+
+export const PathCell = ({ webSiteItem }: { webSiteItem: WebsiteType }) => {
+  const hostingUrl = useHostingUrlWithMultisite(
+    webSiteItem?.currentState.hosting.serviceName,
+    true,
+  );
+
+  return (
+    <Links
+      label={webSiteItem?.currentState.path}
+      href={hostingUrl}
+      target="_blank"
+    />
+  );
+};
+
+export const ServiceNameCell = ({
+  webSiteItem,
+}: {
+  webSiteItem: WebsiteType;
+}) => {
+  const hostingUrl = useHostingUrlWithMultisite(
+    webSiteItem?.currentState.hosting.serviceName,
+  );
+
+  return (
+    <Links
+      label={webSiteItem?.currentState.hosting.serviceName}
+      href={hostingUrl}
+      target="_blank"
+    />
+  );
+};
+
+export const DisplayNameCell = ({
+  webSiteItem,
+}: {
+  webSiteItem: WebsiteType;
+}) => {
+  const hostingUrl = useHostingUrlWithMultisite(
+    webSiteItem?.currentState.hosting.serviceName,
+  );
+
+  return (
+    <Links
+      label={webSiteItem?.currentState.hosting.displayName}
+      href={hostingUrl}
+      target="_blank"
+    />
+  );
+};
+
```

Comment:
I have impression that is some duplicate of code in these function "cell" because there is only the label that is differente